python/triton/language/core.py

```diff
@@ -752,9 +753,19 @@ def to(self, dtype, bitcast=False, _builder=None):
         return semantic.cast(self, dtype, _builder)
 
 
+class constexpr_placeholder(tensor):
```

Comment:
constexpr_placeholder helps to make `tl.constexpr` as normal `tl.tensor`. It is possible to use the `tl.tensor` directly, but here we still hope to make it as a constexpr.